core/embed/sec/tropic/tropic.c

```diff
@@ -53,6 +54,50 @@ static bool tropic_get_tropic_pubkey(lt_handle_t *handle,
                                      curve25519_key pubkey);
 #endif
 
+static bool session_start(tropic_driver_t *drv,
+                          pkey_index_t pairing_key_index) {
+  curve25519_key trezor_private = {0};
+  if (pairing_key_index == TROPIC_FACTORY_PAIRING_KEY_SLOT) {
+    tropic_get_factory_privkey(trezor_private);
+  }
+  if (pairing_key_index == TROPIC_PRIVILEGED_PAIRING_KEY_SLOT) {
+    if (secret_key_tropic_pairing_privileged(trezor_private) != sectrue) {
+      goto cleanup;
+    }
+  }
+  if (pairing_key_index == TROPIC_UNPRIVILEGED_PAIRING_KEY_SLOT) {
+    if (secret_key_tropic_pairing_unprivileged(trezor_private) != sectrue) {
+      goto cleanup;
+    }
+  }
+
+  curve25519_key trezor_public = {0};
+  curve25519_scalarmult_basepoint(trezor_public, trezor_private);
+
+  curve25519_key tropic_public = {0};
+  if (secret_key_tropic_public(tropic_public) != sectrue) {
+#if !PRODUCTION
+    if (!tropic_get_tropic_pubkey(&drv->handle, tropic_public))
+#endif
+    {
+      goto cleanup;
+    }
+  }
+
+  if (lt_session_start(&drv->handle, tropic_public, pairing_key_index,
+                       trezor_private, trezor_public) != LT_OK) {
+    goto cleanup;
+  }
+
+  drv->pairing_key_index = pairing_key_index;
+  drv->initialized = true;
+
+cleanup:
+  memzero(trezor_private, sizeof(trezor_private));
+
+  return drv->initialized;
```

Comment:
If we use `drv->initialized` as the return value, then we should set it to `false` at the beginning of the function.

Reply:
Fixed in https://github.com/trezor/trezor-firmware/pull/5845/commits/6d5f79ee24f2f275c0f088b4424283d353f6435a.